### PR TITLE
fetch suggestions

### LINKED ISSUE #283: JSDoc to type annotation codefix not offered
## Checklist

- [x] I have searched both open and closed issues and cannot find a duplicate.
- [x] I can [reproduce the problem](https://github.com/ananthakumaran/tide/wiki/Explanations-about-the-Issue-Reporting-Checklist#use-the-latest-version-of-the-relevant-packages) with the latest version of the relevant packages.
- [x] The problem still occurs after I issued [`M-x tide-restart-server`](https://github.com/ananthakumaran/tide/wiki/Explanations-about-the-Issue-Reporting-Checklist#try-m-x-tide-restart-server) in the buffer where I had the problem.
- [x] I verified that the version and the configuration file path reported by [`M-x tide-verify-setup`](https://github.com/ananthakumaran/tide/wiki/Explanations-about-the-Issue-Reporting-Checklist#check-the-information-reported-by-m-x-tide-verify-setup) are correct.
- [x] I am 0.8 certain the problem does not belong to `typescript-mode` or `tsserver`. Tests all pass for my local build, but I haven't looked at the tsserver log after requesting codefixes.

## Relevant Version Numbers

- Tide: tide-20180608.239 and tide-20180810.403
- TypeScript: 3.1 and recent builds of 3.2
- Emacs: 25.2.2

## Steps to Reproduce the Bug

Start with a TS file that has JSDoc types:

```ts
// @Filename: welove.ts
/**
 * @param {String | Number} value
 */
export const f = value => new Number(value)
```

with a tsconfig that has noImplicitAny on:

```json
{
    "compilerOptions": {
        "strict": true,
        "noImplicitAny": true,
        "target": "es5",
        "lib": ["es5", "es2015", "es2016", "es2017", "dom"],
        "allowJs": true,
        "checkJs": true,
        "outDir": "out",
        "jsx": "preserve",
        "esModuleInterop": true,
        "module": "commonjs"
    },
    "include": [
        "ex.d.ts",
        "mod1.js",
        "test.js",
        "welove.ts",
        "first.ts",
        "forever.js"
    ]
}
```

## Expected behavior:

Implicit any error on `value`, plus a codefix that replaces it with `value: string | number`

## Actual behavior:

Error, but no codefix offered.

This codefix works fine on VS Code, and other codefixes work fine, such as creating property declarations. I'm not sure what is different about this one.

Edit: related issue https://github.com/Microsoft/TypeScript/issues/28084

### PR DESCRIPTION
**Pull request by ananthakumaran**
fixes #283 

Suggestions from tsserver are shown as flycheck **info** messages. This fix also enables other types of codefixes like convert to an async function, value is never read etc.

screen shot 2018-10-24 at 9 38 16 am

### CODE REVIEW

File: tide.el [hunk 1]
```diff
@@ -2020,13 +2035,16 @@ code-analysis."
     (let ((inhibit-read-only t))
       (erase-buffer))
     (display-buffer (current-buffer) t)
-    (let* ((project-files (-filter (lambda (file-name)
-                                     (not (string-match-p "node_modules/typescript/" file-name)))
+    (let* ((project-files (-reject (lambda (file-name)
+                                     (or (string-match-p "node_modules/typescript/" file-name)
+                                         (string-match-p "tsconfig.json$" file-name)))
```
Comment (josteink) [inline tide.el]: Any reason we need to add this additional filter? It's not obvious from the context.

**Comment by josteink** [comment]
This is a really nice addition!

Tested and looks good to me.

File: tide.el (hunk 1 above)
Comment (ananthakumaran) [reply tide.el]: The file list returned by tsserver also contains tsconfig.json -- not sure when this change was introduced. The callback was waiting for diagnostics results of tsconfig.json which would never arrive.

Comment (josteink) [reply tide.el]: Right. Not sure if that needs a comment or not.

Otherwise, the changes looks fine, and I'm happy about the improvements.

**Pull request merged**